stdlib/std.jsonnet

```diff
@@ -1041,7 +1042,43 @@ limitations under the License.
         std.join('', lines);
     aux(value, [], ''),
 
-  manifestYamlDoc(value, indent_array_in_object=false)::
+  manifestYamlDoc(value, indent_array_in_object=false, quote_keys=true)::
+    local isReserved(key) =
+      local reserved = ['true', 'false', 'null'];
+      local bad = [word for word in reserved if word == key];
+      if std.length(bad) > 0 then
+        true
+      else false;
+    local bareSafe(key) =
+      # Letters has '_' and '-' because they can be treated equally
+      local letters = std.set(std.stringChars('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_-'));
+      local numbers = std.set(std.stringChars('0123456789'));
+      local floatChars = std.set(numbers + std.stringChars('.'));
+      local all_good = std.set(letters + floatChars);
+      local keyChars = std.stringChars(key);
+      local keySet = std.set(keyChars);
+      // Is reserved word
+      if isReserved(key) then
+        false
+      // Consists of only letters
+      else if std.length(std.setInter(letters, keySet)) == std.length(keySet) then
+        // Since this isn't a reserved word, we know this safe
+        true
+      // Unsafe characters
+      else if std.length(std.setInter(all_good, keySet)) != std.length(keySet) then
+        false
+      // Only numbers
+      else if std.length(std.setInter(numbers, keySet)) == std.length(keySet) then
```

Comment:
Unquoted scalars like `-1_000`, `0x1`, and `0o1` are integers according to v1.2 spec. Psych also recognizes base-2 integers, e.g. `0b1`

Reply:
Both hexadecimal and binary are already supported.

According to the spec you linked, octal is supported but the `0o1` notation is not.

What i did miss was negative numbers of all formats.  I will have an update to this shortly to correct that.

Reply:
Negative numbers of all types are now supported and quoted, with extensive tests that cover every individual permutation as well a significant number of combinatory permutations.